### ISSUE TITLE
headless firefox + marionette + gecko + selenium - fails to open web driver intermittently when ran in parallel

### ISSUE THREAD

**Issue by nddipiazza** [mozilla/geckodriver#1135]
## System
* Version: 0.19.1
* Platform: Linux ubuntu 17
* Firefox: 59.0
* Selenium: 3.5.3

## Testcase

Run many selenium + firefox headless sessions in parallel. Run them until you see a failure. 

java log: `Caused by: org.openqa.selenium.WebDriverException: EOF reading marionette message`

gecko log: `<- 500 Internal Server Error {"value":{"error":"unknown error","message":"Failed to decode response from marionette","`

In my case, i could reproduce with about 15 headless remote web driver's running in parallel. 

## Stacktrace

java selenium error: 

```
2018-01-17T13:38:47,128 - INFO  [ae2c2198713a40f9aa6c60e807b47519-fetcher-1:Crawler$FetchCallable@395] - {} - fetch() failure: 
	id: http://localhost:39573/resources/scrubbed/web/test-page/js-test-pages/jsPage_2.html
	parentID: http://localhost:39573/resources/scrubbed/web/test-page/js-test-pages/jsPage_1.html
	batchID: 1
	depth: 1
	fetchedDate: Wed Jan 17 13:38:44 CST 2018
	lastModified: Wed Dec 31 18:00:00 CST 1969
	contentSignature: null
	signature: 0
	linked: true
	discarded: false
	errorCount: 1
	error:
		crawler.common.CrawlItemException: phase=FETCH; This item failed during fetch()
			at crawler.Crawler$FetchCallable.call(Crawler.java:394)
			at crawler.Crawler$FetchCallable.call(Crawler.java:362)
			at java.util.concurrent.FutureTask.run(FutureTask.java:266)
			at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
			at java.util.concurrent.FutureTask.run(FutureTask.java:266)
			at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
			at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
			at java.lang.Thread.run(Thread.java:748)
		Caused by: org.openqa.selenium.WebDriverException: EOF reading marionette message
		Build info: version: '3.5.3', revision: 'a88d25fe6b', time: '2017-08-29T12:42:44.417Z'
		System info: host: 'ubuntu17mate', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.13.0-25-generic', java.version: '1.8.0_151'
		Driver info: driver.version: RemoteWebDriver
		remote stacktrace: stack backtrace:
		   0:           0x4edb3c - backtrace::backtrace::trace::hc4bd56a2f176de7e
		   1:           0x4edb72 - backtrace::capture::Backtrace::new::he3b2a15d39027c46
		   2:           0x4e9586 - webdriver::error::WebDriverError::new::hee9ff9dd0a444441
		   3:           0x4e9e17 - <webdriver::error::WebDriverError as core::convert::From<std::io::error::Error>>::from::h9114c5c53ee93e22
		   4:           0x4477c0 - geckodriver::marionette::MarionetteHandler::create_connection::hf0532ddb9e159684
		   5:           0x428570 - <webdriver::server::Dispatcher<T, U>>::run::h2119c674d7b88193
		   6:           0x4029b9 - std::sys_common::backtrace::__rust_begin_short_backtrace::h21d98a9ff86d4c25
		   7:           0x40be65 - std::panicking::try::do_call::h5cff0c9b18cfdbba
		   8:           0x5e6a6c - panic_unwind::__rust_maybe_catch_panic
		                        at /checkout/src/libpanic_unwind/lib.rs:99
		   9:           0x41eb22 - <F as alloc::boxed::FnBox<A>>::call_box::h413eb1d9d9f1c473
		  10:           0x5df13b - alloc::boxed::{{impl}}::call_once<(),()>
		                        at /checkout/src/liballoc/boxed.rs:692
		                         - std::sys_common::thread::start_thread
		                        at /checkout/src/libstd/sys_common/thread.rs:21
		                         - std::sys::imp::thread::{{impl}}::new::thread_start
		                        at /checkout/src/libstd/sys/unix/thread.rs:84
			at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
			at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
			at sun.reflect.DelegatingConstructorAccessorImpl.__newInstance(DelegatingConstructorAccessorImpl.java:45)
			at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java)
			at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java)
			at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
			at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$new$0(W3CHandshakeResponse.java:57)
			at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$getResponseFunction$2(W3CHandshakeResponse.java:104)
			at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$24(ProtocolHandshake.java:359)
			at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
			at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
			at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
			at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:498)
			at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
			at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
			at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
			at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
			at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:464)
			at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:362)
			at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:136)
			at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:142)
			at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:646)
			at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:255)
			at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:237)
			at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:138)
			at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:175)
			at crawler.fetch.impl.script.factory.FirefoxFactoryImpl.getDriver(FirefoxFactoryImpl.java:33)
			
```

## Trace-level log

geckodriver Level.ALL log: https://pastebin.com/raehqACZ

**Comment by whimboo** [comment]
Would you mind running a debug build of Firefox so we can get more output in the trace log? You can find a recent 59.0a1 build here:

https://queue.taskcluster.net/v1/task/fQ3U-Y2tQsCXHRx1hopD6w/runs/0/artifacts/public/build/target.tar.bz2

Please note that geckodriver in version 0.19.1 still crashes Firefox during shutdown. This is fixed by https://bugzilla.mozilla.org/show_bug.cgi?id=1403923, but hasn't been released yet. For testing you could use the file `bin/geckodriver` from our common tests archive matching the above build:

https://queue.taskcluster.net/v1/task/fQ3U-Y2tQsCXHRx1hopD6w/runs/0/artifacts/public/build/target.common.tests.zip

**Comment by whimboo** [comment]
@leobalter, if you have a problem please open your own issue but don't hijiack old ones. I will hide your comment on this issue now.

Closing this issue because of no response from @nddipiazza.

**Comment by leobalter** [comment]
I did my best to avoid filing a duplicate issue, and searched a lot on the issues until I found this one.

I’m not sure why you are accusing me of hijacking issues, but it was certainly not my intention neither it was clear to me that your triage disregards 8 months old issues. That’s not really long, but that’s a personal opinion.

It’s frustrating to see this is the only feedback for my report where I tried to provide a good amount of info so we could work on a fix. I’d be happy to provide more info and eventually filing a patch if I had some help to learn more.

Seeing my comment flagged as off-topic and this issue closed with no further action is nothing but frustrating. From this point, I hope the maintainers could have this eventually fixed, but I spent my energy here already. I’m volunteering to not contribute to this project anymore while your policy remains the same. Sorry for this, but the burnout is high on OSS and I can’t handle it.
